qutebrowser/misc/tabsfinder.py

```diff
@@ -0,0 +1,43 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+import re
+from collections import namedtuple
+
+from qutebrowser.utils import objreg
+
+Tab = namedtuple('Tab', 'win_id index url title')
+
+
+def all_tabs(skip_win_id=None):
+    for win_id in objreg.window_registry:
+        tabbed_browser = objreg.get('tabbed-browser', scope='window', window=win_id)
+
+        if skip_win_id is not None and win_id == skip_win_id:
+            continue
+
+        if tabbed_browser.shutting_down:
+            continue
+
+        for idx in range(tabbed_browser.widget.count()):
+            tab = tabbed_browser.widget.widget(idx)
+            t = Tab(int(win_id), idx + 1, tab.url(),
+                    tabbed_browser.widget.page_title(idx))
+            yield(t)
+
+
+def delete_tab(data):
+    """Close the selected tab."""
+
+    pattern = "^\d+/\d+$"
+
+    if re.match(pattern, data[0]):
+        index = data[0]
+    elif re.match(pattern, data[2]):
+        index = data[2]
```

Comment:
This seems weird. Why use a regex to try and find something to close? Instead of having to guess, the caller should tell the function where to find the ID (for example by using [functools.partial](https://docs.python.org/3/library/functools.html#functools.partial) to get a function usable for `delete_func`).